### PR TITLE
Add argument group

### PR DESCRIPTION
**Pull request by pavalos6401**
Here is my first attempt at adding groups. 

This is to address issue #48 

If there is more to be done for this pull request, please let me know.

### CODE REVIEW

File: argparse_dataclass.py [hunk 1]
```diff
@@ -416,7 +416,32 @@ def _add_dataclass_options(
                         "For Union types other than 'Optional', a custom 'type' must be specified using "
                         "'metadata'."
                     )
-        parser.add_argument(*args, **kwargs)
+
+        if "group" in field.metadata:
```
Comment (mivade) [inline argparse_dataclass.py:420]: Can this be made into a method and unit tested?

**Review by mivade — COMMENTED** [review]
Looks good to start with. In addition to some unit testing could you please add a usage example?

**Comment by pavalos6401** [comment]
Hey, I've moved the code to its own function and cleaned it up a bit. I also added documentation in the readme for some simple usage examples. I'll get started on the unit tests, but I might not get to it till tomorrow.

**Comment by pavalos6401** [comment]
I have added tests. Let me know if they make sense, or if you expected a different way to test them.

**Comment by mivade** [comment]
Test failures appear to be related to formatting. If you run black against the code that should resolve.

**Comment by pavalos6401** [comment]
I've run black now, it seems it was one line that was too long.

**Review by mivade — APPROVED** [review]

**Comment by mivade** [comment]
Thanks!

**Pull request merged**